src/fetchers/stats-fetcher.js

```diff
@@ -63,37 +63,57 @@ const fetcher = (variables, token) => {
   );
 };
 
+const fetchYearCommits = (variables, token) => {
+  return request({
+    query: `
+      query userInfo($login: String!, $from_time: DateTime!) {
+        user(login: $login) {
+          contributionsCollection(from: $from_time) {
+            totalCommitContributions
+            restrictedContributionsCount
+          }
+        }
+      }
+      `, variables,
+  }, {
+    Authorization: `bearer ${token}`,
+  },);
+};
+
 // https://github.com/anuraghazra/github-readme-stats/issues/92#issuecomment-661026467
 // https://github.com/anuraghazra/github-readme-stats/pull/211/
-const totalCommitsFetcher = async (username) => {
+const totalCommitsFetcher = async (username, contributionYears) => {
   if (!githubUsernameRegex.test(username)) {
     logger.log("Invalid username");
     return 0;
   }
 
-  // https://developer.github.com/v3/search/#search-commits
-  const fetchTotalCommits = (variables, token) => {
-    return axios({
-      method: "get",
-      url: `https://api.github.com/search/commits?q=author:${variables.login}`,
-      headers: {
-        "Content-Type": "application/json",
-        Accept: "application/vnd.github.cloak-preview",
-        Authorization: `token ${token}`,
-      },
-    });
-  };
+  let totalPublicCommits = 0;
+  let totalPrivateCommits = 0;
 
   try {
-    let res = await retryer(fetchTotalCommits, { login: username });
-    if (res.data.total_count) {
-      return res.data.total_count;
-    }
+    await Promise.all(contributionYears.map(async (year) => {
+          let variables = {
+            login: username,
+            from_time: `${year}-01-01T00:00:00.000Z`,
+          };
+          let res = await retryer(fetchYearCommits, variables);
+          totalPublicCommits += res.data.data.user.contributionsCollection.totalCommitContributions;
+          totalPrivateCommits += res.data.data.user.contributionsCollection.restrictedContributionsCount;
+        })
+    );
+    return {
+      totalPublicCommits,
+      totalPrivateCommits,
+    };
   } catch (err) {
     logger.log(err);
     // just return 0 if there is something wrong so that
     // we don't break the whole app
-    return 0;
+    return {
+      totalPublicCommits: 0,
+      totalPrivateCommits: 0,
```

Comment:
Maybe we can also return a success boolean and use it in 

https://github.com/anuraghazra/github-readme-stats/blob/c03bb2f250f6d65802ca16101aa38baef1022a5c/api/index.js#L66

To set the header?

Reply:
I choose to throw an error if any error occurs, just like what `fetchStats()` and `totalStarsFetcher()` do.

Reply:
Great! For your case, that is the best option since your using the GraphQL API!